File: Data Structures & Algorithms/validate-parentheses/submission-6.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        close2open = {")": "(", "}" : "{", "]" : "["}

        for c in s:
            if c in close2open:
                if stack and stack[-1] == close2open[c]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(c)
        
        return True if not stack else False
            



        

File: Data Structures & Algorithms/car-fleet/submission-6.kt
class Solution {
    fun carFleet(target: Int, position: IntArray, speed: IntArray): Int {

        val stack = ArrayDeque<Double>()

        val combined = Array(position.size) { i -> Pair(position[i], speed[i])}

        val sortedComb = combined.sortedBy { it.first }

        for (car in sortedComb.reversed()) {
            val time: Double = (target - (car.first)).toDouble() / (car.second).toDouble()

            if (stack.isNotEmpty() && time <= stack.last()) continue

            else stack.addLast(time)
        }
        return stack.size
    }
}
